### PR TITLE
fix(cli): align truncated flag with content deduplication

### PR DESCRIPTION
**Pull request by TimeToBuildBob**
## Summary

Follow-up fixes from Greptile review of PR #20 (scored 4/5):

- **Fix truncated/results_in_context inconsistency**: When `assemble_context` deduplicates chunks (identical content from different sources), `truncated` now correctly reflects the drop. Previously `truncated=False` with `results_in_context < total_results` was possible, confusing machine consumers.
- **Remove dead `_count_tokens` alias**: All callers now use the public `count_tokens` method directly. The private alias with its misleading comment was dead code.
- **Add dedup test**: `test_search_json_truncated_on_dedup` exercises the duplicate-content path and verifies the invariant holds.

## Test plan

- [x] All 10 CLI search tests pass (including new dedup test)
- [x] Existing invariant test (`test_search_json_output_context_info` lines 135-136) continues to pass

### CODE REVIEW

**Comment by TimeToBuildBob** [comment]
@greptileai review

File: tests/test_cli_search.py [hunk 1]
```diff
@@ -243,6 +243,61 @@ def test_search_json_expand_truncated_consistency(populated_index):
...
+    if data["total_results"] > 1 and ctx["results_in_context"] < data["total_results"]:
+        # The dedup dropped results — truncated must be True
+        assert ctx["truncated"] is True, (
+            f"results_in_context={ctx['results_in_context']} < "
+            f"total_results={data['total_results']} but truncated=False"
+        )
+    # The invariant from test_search_json_output_context_info still holds:
+    # if not truncated, results_in_context == total_results
+    if not ctx["truncated"]:
+        assert ctx["results_in_context"] == data["total_results"]
```
[the remark this answers is not in the document]
Comment (TimeToBuildBob) [reply tests/test_cli_search.py]: Good catch. Fixed in 465fb6c — the test now asserts unconditionally that both docs are returned by ChromaDB (they have distinct IDs so the indexer stores both) and that `assemble_context` drops one via content-dedup, making `results_in_context == 1`. Verified locally: test passes with the concrete assertions.

**Comment by TimeToBuildBob** [comment]
@greptileai review

**Comment by TimeToBuildBob** [comment]
PR is ready to merge:
- ✅ CI: all 3 tests passing (3.10, 3.11, 3.12)
- ✅ Greptile: 5/5 confidence — "safe to merge"
- ✅ No unresolved review threads

Needs maintainer merge (no push access to gptme-rag).
